packages/node_modules/@ciscospark/react-container-call-incoming/src/index.js

```diff
@@ -0,0 +1,128 @@
+import {Component, PropTypes} from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import {Howl} from 'howler';
+
+import {
+  callIncoming,
+  listeningForIncomingCalls,
+  removeCallIncoming
+} from '@ciscospark/redux-module-media';
+
+import {
+  NOTIFICATION_TYPE_CALL,
+  createNotification
+} from '@ciscospark/react-container-notifications';
+
+import Ringtone from './media/Ringtone.mp3';
+
+
+export class IncomingCall extends Component {
+  constructor(props) {
+    super(props);
+    this.ringtone = new Howl({
+      src: [Ringtone],
+      loop: true
+    });
+  }
+
+  componentDidMount() {
+    const {media, spark, onEvent} = this.props;
+    if (media.getIn([`status`, `isListening`])) {
+      return;
+    }
+    spark.phone.on(`call:incoming`, (incomingCall) => {
```

Comment:
I actually started off with it in the redux module but changed it over to this after we needed to get current store data regarding media state when a call came in. Take a look at https://github.com/adamweeks/react-ciscospark-1/blob/6ea6a9037577c16ef483c6583a90007d314602b8/packages/node_modules/%40ciscospark/react-container-call-incoming/src/index.js#L86 to see what I'm talking about. 

I just came across the `getState` ability of thunk though. So maybe I can switch over to it. https://github.com/gaearon/redux-thunk#motivation